Pass AuxilaryAuthHeader for referenced resourceIds during Put Connection.

Nilambari opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

notyashhh commented:
/azp run

notyashhh commented:
@Nilambari, a few things,

1. please update the changelog
2. make sure you have written test cases for the cmdlets changed, and use the added/removed parameters. (Add any new tests for scenarios that are relevant)
3. There are about 50 Static Analysis errors in this PR due to the use of `--` symbols in the help files. (Please replace them with something else) (Exact Error: "The '--' operator works only on variables or on properties.")


Let's fix these then I can point out more, you can also directly view them here: [logs](https://dev.azure.com/azclitools/public/_build/results?buildId=226535&view=logs&j=b9afb22f-0837-5d02-bc24-6c30aee08bcf&t=a0002f0d-5a71-5c30-091a-1fe5252e1c8a&l=1339)

Nilambari commented:
> @Nilambari, a few things,
> 
> 1. please update the changelog
> 2. make sure you have written test cases for the cmdlets changed, and use the added/removed parameters. (Add any new tests for scenarios that are relevant)
> 3. There are about 50 Static Analysis errors in this PR due to the use of `--` symbols in the help files. (Please replace them with something else) (Exact Error: "The '--' operator works only on variables or on properties.")
> 
> Let's fix these then I can point out more, you can also directly view them here: [logs](https://dev.azure.com/azclitools/public/_build/results?buildId=226535&view=logs&j=b9afb22f-0837-5d02-bc24-6c30aee08bcf&t=a0002f0d-5a71-5c30-091a-1fe5252e1c8a&l=1339)

@notyashhh 
1. Updated changelog.
2. There is no new param added, so no tests changes needed. We already have tests coverage.
3. Fixed Static Analysis error by updating help files. Will check next build to confirm they are fixed. Thanks!

Nilambari commented:
/azp run

notyashhh commented:
/azp run

notyashhh commented:
@Nilambari, Thanks for resolving these!
 
Also, Thanks for catching the `CreateOrUpdate` -> `CreateOrUpdateWithHttpMessagesAsync` Issue.

There is 1 more issue remaining that relates to the file modified in Network.Management.Sdk folder.
This folder cannot be manually edited for any code changes (as this is directly generated by swagger)

Can you please revert the Management.Sdk changes in [VirtualNetworkGatewayConnectionsOperationsExtensions.cs](https://github.com/Azure/azure-powershell/pull/27310/files#diff-8f14cbb423ed75fd23b0a020e76d36a1ec198d115b01e7020551bba60b0ff86c) ?

Nilambari commented:
As per discussion, opened new PR: https://github.com/Azure/azure-powershell/pull/27349 targeting to main branch. Thanks!
